site/app/entities/forum/Post.php

```diff
@@ -63,9 +104,80 @@ class Post {
     #[ORM\Column(type: Types::BOOLEAN)]
     protected bool $render_markdown;
 
+    public function isRenderMarkdown(): bool {
+        return $this->render_markdown;
+    }
+
     /**
      * @var Collection<PostHistory>
      */
     #[ORM\OneToMany(mappedBy: "post", targetEntity: PostHistory::class)]
     protected Collection $history;
+
+    /**
+     * @return Collection<PostHistory>
+     */
+    public function getHistory(): Collection {
+        return $this->history;
+    }
+
+    /**
+     * @var Collection<PostAttachment>
+     */
+    #[ORM\OneToMany(mappedBy: "post", targetEntity: PostAttachment::class)]
+    protected Collection $attachments;
+
+    /**
+     * @return Collection<PostAttachment>
+     */
+    public function getAttachments(): Collection {
+        return $this->attachments;
+    }
+
+    /**
+     * @var Collection<UserEntity>
+     */
+    #[ORM\ManyToMany(targetEntity: UserEntity::class, inversedBy: "upducks")]
+    #[ORM\JoinTable(name: "forum_upducks")]
+    #[ORM\JoinColumn(name: "post_id", referencedColumnName:"id")]
+    #[ORM\InverseJoinColumn(name:"user_id", referencedColumnName:"user_id")]
+    protected Collection $upduckers;
+
+    /**
+     * @return Collection<UserEntity>
+     */
+    public function getUpduckers(): Collection {
+        return $this->upduckers;
+    }
+
+    protected int $reply_level = 1;
+    public function getReplyLevel(): int {
+        return $this->reply_level;
+    }
+    public function setReplyLevel(int $new): void {
+        $this->reply_level = $new;
+    }
+    /**
+     * Doctrine ORM does not use constructors, instead filling properties from database.
+     * We are free to make constructors for "empty" or "junk" posts.
+     */
+
+    public function __construct(Thread $merged_thread) {
+        $this->content = '';
+        $this->render_markdown = false;
+        $this->author = $merged_thread->getAuthor();
+        $this->thread = $merged_thread;
+        $this->deleted = false;
+        $this->anonymous = true;
+        $this->id = -1;
+    }
```

Comment:
Is there a reason why this shouldn't accept parameters for each field, similar to how the Poll entity handles things?

Reply:
This constructor is meant for when a thread has no posts (because it's merged) and we need an "empty" post to serve as its first post. Several properties of the first post of a thread get dereferenced when rendering a thread, so we want to make sure we have enough empty values so as to not break things.

When implementing CRUD operations down the road, we may want more constructors that accept these parameters, but I wrote this one so the View method doesn't have to think beyond "I need a blank post for this thread so I don't crash".